apps/storefront/src/pages/OrderDetail/components/OrderBilling.tsx

```diff
@@ -105,13 +166,21 @@ export default function OrderBilling({ isCurrentCompany }: OrderBillingProps) {
             </Box>
 
             <B3ProductList
-              products={billingItem.digitalProducts}
+              products={digitalProducts.length ? digitalProducts : billingItem.digitalProducts}
               totalText="Total"
               canToProduct={isCurrentCompany}
               textAlign={isMobile ? 'left' : 'right'}
               money={money}
+              getDigitalDownloadLinks={getCurrentDigitalProduct}
             />
           </CardContent>
+          <DownloadDigitalProductsDialog
+            isOpen={isDigitalDownloadOpen}
+            onClose={() => setIsDigitalDownloadOpen(false)}
+            product={currentDigitalProduct}
+            b3Lang={b3Lang}
+            handleDownloadDigitalFile={handleDownloadDigitalFile}
+          />
```

Comment:
```suggestion
          { currentDigitalProduct && (<DownloadDigitalProductsDialog
            isOpen={isDigitalDownloadOpen}
            onClose={() => setIsDigitalDownloadOpen(false)}
            product={currentDigitalProduct}
            b3Lang={b3Lang}
            handleDownloadDigitalFile={handleDownloadDigitalFile}
          />)}
```

Reply:
@bc-micah @bc-victor Waiting for currentDigitalProduct to be set up when clicking View Files button was causing a delay in the modal creating a bad experience for users. I changed the way we open the modal here, I deleted one of the state variables `isDigitalDownloadOpen` and used currentDigitalProduct to validate if open should be opened or not, so in the dialog the products prop can stay undefined.